### PR TITLE
Drop python 3.7 from starters

### PR DESCRIPTION
**Pull request by SajidAlamQB**
## Motivation and Context

related to: https://github.com/kedro-org/kedro/issues/2158

Dropping python 3.7 support for starters.

## How has this been tested?

Changed `test_requirements.txt` to use develop over main since we are moving to `0.19.0` and it will help unblock develop to use starters with the new add-ons flow.

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Assigned myself to the PR
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by lrcouto — APPROVED** [review]

**Comment by merelcht** [comment]
Great stuff @SajidAlamQB ! We'll need to wait with merging this until `0.18.14` is out though.

File: .circleci/config.yml [hunk 1]
```diff
@@ -206,18 +206,18 @@ workflows:
       - run:
           matrix:
             parameters:
-              python_version: [ "3.7", "3.8", "3.9", "3.10" ]
+              python_version: [ "3.8", "3.9", "3.10" ]
```
Comment (astrojuanlu) [inline .circleci/config.yml:209]: Should we be adding 3.11 here at some point?

Comment (SajidAlamQB) [reply .circleci/config.yml:209]: Yes, we will need to update CI to GitHub actions for it as well.

**Comment by SajidAlamQB** [comment]
It looks like the renamed `_data_sets` attribute in `Catalog` to `_datasets` is causing issues with kedro-viz. `kedro_viz/integrations/kedro/hooks.py` on L34 is referencing `self.datasets = catalog._data_sets` which will need to be updated once 0.19 is out but in the meantime since this PR is trying to use develop branch it will conflict with viz.

**Comment by merelcht** [comment]
> It looks like the renamed `_data_sets` attribute in `Catalog` to `_datasets` is causing issues with kedro-viz. `kedro_viz/integrations/kedro/hooks.py` on L34 is referencing `self.datasets = catalog._data_sets` which will need to be updated once 0.19 is out but in the meantime, since this PR aims to switch to develop branch, it will conflict with viz.

I've opened this PR https://github.com/kedro-org/kedro-viz/pull/1587 to fix that 🙂

**Review by merelcht — APPROVED** [review]
LGTM! Do we need a reminder ticket to set the `kedro-viz` and `kedro` requirements back to what they were?

**Pull request merged**